java/osmosis/Portfile

```diff
@@ -0,0 +1,122 @@
+
+patchfiles          package-bin-osmosis.diff \
+                    build-gradle.diff
+
+post-patch {
+    reinplace "s|/etc/osmosis|${confdir}/${name}.conf|g" \
+        ${worksrcpath}/package/bin/osmosis
+    reinplace "s|@PREFIX@|${prefix}|g" \
+        ${worksrcpath}/package/bin/osmosis
+    reinplace "s|@VERSION@|${version}|g" \
+        ${worksrcpath}/build.gradle
+    reinplace "s|/usr/share|${prefix}/share|g" \
+        ${worksrcpath}/package/script/munin/README
+    reinplace "s|/etc|${prefix}/etc|g" \
+        ${worksrcpath}/package/script/munin/README
+    reinplace "s|#user osm|#user nobody|g" \
+        ${worksrcpath}/package/script/munin/osm_replication.conf
+    reinplace "s|#env.osmosis /opt/osmosis/bin|env.osmosis ${appdir}/bin|g" \
+        ${worksrcpath}/package/script/munin/osm_replication.conf
+    reinplace "s|#env.workingDirectory /path/to/state.txt|#env.workingDirectory ${prefix}/var/lib/mod_tile/.osmosis/state.txt|g" \
+        ${worksrcpath}/package/script/munin/osm_replication.conf
+}
+
+destroot {
+    xinstall -m 0755 -d ${destroot}/${appdir}
+    xinstall -m 0755 -d ${destroot}/${appdir}/bin
+    xinstall -m 0755 -d ${destroot}/${appdir}/config
+    xinstall -m 0755 -d ${destroot}/${appdir}/lib
+    xinstall -m 0755 -d ${destroot}/${appdir}/lib/default
+    xinstall -m 0755 -d ${destroot}/${appdir}/script
+    xinstall -m 0755 -d ${destroot}/${appdir}/script/contrib
+    xinstall -m 0755 -d ${destroot}/${appdir}/script/munin
+    xinstall -m 0755 -d ${destroot}/${confdir}
+    xinstall -m 0755 -d ${destroot}/${docdir}
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/*.txt] \
+        ${worksrcpath}/README.md \
+        ${destroot}${docdir}
+    xinstall -m 0755 ${worksrcpath}/package/bin/osmosis \
+        ${destroot}${appdir}/bin
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/config/*] \
+        ${destroot}${appdir}/config
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/lib/default/*] \
+        ${destroot}${appdir}/lib/default
+    xinstall -m 0755 \
+        {*}[glob ${worksrcpath}/package/script/*.sh] \
+        ${destroot}${appdir}/script
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/script/*.sql] \
+        ${destroot}${appdir}/script
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/script/*.txt] \
+        ${destroot}${appdir}/script
+    xinstall -m 0755 \
+        {*}[glob ${worksrcpath}/package/script/contrib/*.sh] \
+        ${destroot}${appdir}/script/contrib
+    xinstall -m 0644 \
+        {*}[glob ${worksrcpath}/package/script/contrib/*.sql] \
+        ${destroot}${appdir}/script/contrib
+    xinstall -m 0755 \
+        ${worksrcpath}/package/script/munin/osm_replication_lag \
+        ${destroot}${appdir}/script/munin
+    xinstall -m 0644 \
+        ${worksrcpath}/package/script/munin/README \
+        ${worksrcpath}/package/script/munin/osm_replication.conf \
+        ${destroot}${appdir}/script/munin
+}
```

Comment:
same here as in the other port; can this be simplified? Assuming that there is not proper "install" command... if you need all the files you can probably just use `file copy`?

Reply:
I've simplified the port as you suggested.

I also realised it has some tests, so I've enabled the test phase.  Some of the tests fail because they expect to be able to connect to databases.  I'm afraid I've had to disable the tests by deleting them from their source folders as the unit test mechanism works on finding the tests and running them.  Unfortunately, there is no summary output of the tests, with the results all being written to XML files.  Also, if any of the tests fails, it aborts any remaining tests, so removing them seems the best option overall.

I haven't added runtime dependencies on the databases as their usage is optional.